static/app/components/searchQueryBuilder/hooks/useQueryBuilderGrid.tsx

```diff
@@ -88,8 +88,11 @@ export function useQueryBuilderGrid({
       onKeyDownCapture: noop,
       onKeyDown,
       onFocus: (e: FocusEvent) => {
-        // If this element gets focused and there is a selection, focus the SelectionKeyHandler instead
-        if (e.target === ref.current && state.selectionManager.selectedKeys.size > 0) {
```

Comment:
The old logic looked at the selected keys state which was sometimes stale. We know that we never want this element to take focus away from the selection key handler, so we just look at the focus event instead to refocus the right element.